pkg/controller/controller.go

```diff
@@ -81,19 +82,20 @@ var (
 	ttls       = []string{"15", "30", "60", "120", "240"}
 )
 
-func New(h *render.Renderer) *Controller {
+func New(h *render.Renderer, allowlist []string) *Controller {
 	return &Controller{
-		h: h,
+		h:         h,
+		allowlist: allowlist,
 	}
 }
 
-func (c *Controller) HandlePopup(allowlist []string) http.Handler {
+func (c *Controller) HandlePopup() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.Method {
 		case http.MethodGet:
 			c.handlePopupGet(w, r)
 		case http.MethodPost:
-			c.handlePopupPost(w, r, allowlist)
+			c.handlePopupPost(w, r, c.allowlist)
```

Comment:
The allowlist is already dangled off the controller, so you don't need it as a parameter here. It's accessible within `handlePopupPost` off of `c`.

Reply:
ah of course